### PR TITLE
Bump python3 to 3.12.7

### PR DESCRIPTION
**Pull request by Akendo**
Update Python3.12 to the latest release.

### CODE REVIEW

File: prepare_binary [hunk 1]
```diff
@@ -1 +1,4 @@
 DEB_BUILD_OPTIONS="$DEB_BUILD_OPTIONS nobench nolto nopgo"
+# We need libgdbm-compat-dev for x86 or the build process fails will fail 
+# despite libdb beging disabled.
```
Comment (fwilhe) [inline prepare_binary]: what's the relation to libdb?

https://packages.debian.org/sid/libgdbm-compat-dev https://www.gnu.org.ua/software/gdbm/

this seems an unrelated file-based db thing to me

but why do we need this dependency explicitly?

File: prepare_binary [hunk 2]
```diff
@@ -1 +1,4 @@
 DEB_BUILD_OPTIONS="$DEB_BUILD_OPTIONS nobench nolto nopgo"
+# We need libgdbm-compat-dev for x86 or the build process fails will fail
```
Comment (fwilhe) [inline prepare_binary]:
```suggestion
# We need libgdbm-compat-dev for x86 or the build process fails
```

File: prepare_binary (hunk 1 above)
Comment (fwilhe) [inline prepare_binary]:
```suggestion
# despite libdb being disabled.
```

File: prepare_source [hunk 3]
```diff
@@ -1,2 +1,6 @@
-apt_src python3.12
+# 3.12.7 isn't in testing yet.
```
Comment (fwilhe) [inline prepare_source:1]: is there a good reason why we need 3.12.7 now? no objection, just trying to understand the tradeoffs between fixed issues and effort of customized build

File: prepare_binary (hunk 1 above)
Comment (Akendo) [reply prepare_binary]: You're right. I misread that, was thinking that this was related. My bad.

Comment (Akendo) [reply prepare_binary]: I fixed this.

File: prepare_source (hunk 3 above)
Comment (Akendo) [reply prepare_source:1]: It would take some days until the maintainer would push it into testing. The patch was accepted on the [3rd](https://tracker.debian.org/news/1571580/accepted-python312-3127-1-source-into-unstable/) of October and will probably land in testing on the 13th of October. I do not want to wait and get this out of the door now.

**Review by fwilhe — APPROVED** [review]
LTGM, thanks @Akendo
